### PR TITLE
Refactored tasks.

### PR DESCRIPTION
**Pull request by jortel**
Taking a procedural approach to the publishing and syncing tasks in attempt to simplify the plugin.

### CODE REVIEW

File: pulp_file/app/tasks/synchronizing.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+def build_additions(importer, manifest, delta):
+    """
+    Build the content to be added.
+
+    Args:
+        importer (FileImporter): An importer.
+        manifest (Manifest): The downloaded manifest.
+        delta (Delta): The set of Key to be added and removed.
+
+    Yields:
```
Comment (daviddavis) [inline pulp_file/app/tasks/synchronizing.py]: Should this be 'Returns'? I know that it returns an iterable though.

File: pulp_file/app/tasks/synchronizing.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+def build_removals(base_version, delta):
+    """
+    Build the content to be removed.
+
+    Args:
+        base_version (RepositoryVersion):  The base repository version.
+        delta (Delta): The set of Key to be added and removed.
+
+    Yields:
+        SizedIterable: The FileContent to be removed from the repository.
```
Comment (daviddavis) [inline pulp_file/app/tasks/synchronizing.py]: Here too.

**Comment by daviddavis** [comment]
Code looks good. Will test it out.

File: pulp_file/app/tasks/synchronizing.py (hunk 1 above)
Comment (jortel) [reply pulp_file/app/tasks/synchronizing.py]: Yes.  Used to be a generator then changed it to returning the `SizedIterator` and missed the docstring.

File: pulp_file/app/tasks/synchronizing.py (hunk 2 above)
Comment (jortel) [reply pulp_file/app/tasks/synchronizing.py]: Yes.  Used to be a generator then changed it to returning the `SizedIterator` and missed the docstring.

**Pull request merged**